opensrp-opd/src/main/java/org/smartregister/opd/utils/OpdUtils.java

```diff
@@ -125,4 +137,18 @@ public static String convertDate(@NonNull Date date, @NonNull String dateFormat)
         SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateFormat, Locale.US);
         return simpleDateFormat.format(date);
     }
+
+    public static Date convertStringToDate(String pattern, String dateString) {
```

Comment:
Annotate this method with **Nullable**, it can return a null if the `pattern` or `dateString` is incorrect. The better solution would be to let the caller of this `method` handle the exception since they caused it by providing an incorrect `dateString` or `pattern`